ci/build_wheel.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name="cucim"
+package_dir="python/cucim"
+package_src_dir="${package_dir}/src/${package_name}"
+
+CMAKE_BUILD_TYPE="release"
+
+source rapids-configure-sccache
+source rapids-date-string
+
+version=$(rapids-generate-version)
+commit=$(git rev-parse HEAD)
+
+# for CMake VERSION_CPP need to truncate any trailing 'a'
+version_cpp=${version%a*}
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+
+# Patch project metadata files to include the CUDA version suffix and version override.
+pyproject_file="${package_dir}/pyproject.toml"
+
+PACKAGE_CUDA_SUFFIX="-${RAPIDS_PY_CUDA_SUFFIX}"
+# update package name to have the cuda suffix
+sed -i "s/name = \"${package_name}\"/name = \"${package_name}${PACKAGE_CUDA_SUFFIX}\"/g" ${pyproject_file}
+echo "${version}" > VERSION
+echo "${version_cpp}" > VERSION_CPP
+echo "${version_cpp}" > "${package_dir}/VERSION"
+sed -i "/^__git_commit__/ s/= .*/= \"${commit}\"/g" "${package_src_dir}/_version.py"
+
+if [[ ${PACKAGE_CUDA_SUFFIX} == "-cu12" ]]; then
+    # change pyproject.toml to use CUDA 12.x version of cupy
+    sed -i "s/cupy-cuda11x/cupy-cuda12x/g" ${pyproject_file}
+fi
+
+echo "The package name and/or version was modified in the package source. The git diff is:"
+git diff
+
+pip install --upgrade pip
+
+# Install pip build dependencies (not yet using pyproject.toml)
+rapids-dependency-file-generator \
+  --file_key "py_build" \
+  --output "requirements" \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" | tee build_requirements.txt
+pip install -r build_requirements.txt
+
+# Building the libjpeg-turbo dependency requires YASM
+# Also need to install openslide dev libraries on the system
+if ! command -v apt &> /dev/null
+then
+    echo "apt package manager not found, attempting to use yum"
+    yum install yasm openslide-devel -y
+else
+    echo "apt package manager was found"
+    apt update
+    apt install yasm libopenslide-dev -y
+fi
```

Comment:
These dependencies should be installed in the CI image instead: https://github.com/rapidsai/ci-imgs/blob/main/ci-wheel.Dockerfile

Reply:
@raydouglass, I don't think these (yasm and libopenslide-dev) are common to all RAPIDS projects. Do you think we still need to update the umbrella ci-wheel.Dockerfile? In particular, `libopenslide-dev` is cucim-specific and only needed for test cases in C++ while building libcucim (not needed for Python wheel).

Reply:
Yes. They are small packages and relatively independent. We already do this (albeit in a different image & layer) for things like `libnuma-dev`.

Reply:
Adding with PR: https://github.com/rapidsai/ci-imgs/pull/90

Reply:
Next step would be to drop these lines: https://github.com/rapidsai/cucim/pull/619#discussion_r1382176217

Reply:
sorry, merged the suggestion before the CI img update was done. I will reintroduced it for now to check if using NEXT_FULL_TAG as in a comment above is working, but can remove them again later

Reply:
Okay, now there should be no apt/yum calls in the script anymore